aiida/cmdline/groups/dynamic.py

```diff
@@ -12,17 +13,37 @@
 
 from ..params import options
 from ..params.options.interactive import InteractiveOption
-from ..utils import echo
 from .verdi import VerdiCommandGroup
 
 __all__ = ('DynamicEntryPointCommandGroup',)
 
 
 class DynamicEntryPointCommandGroup(VerdiCommandGroup):
-    """Subclass of :class:`click.Group` that loads subcommands dynamically from entry points."""
+    """Subclass of :class:`click.Group` that loads subcommands dynamically from entry points.
 
-    def __init__(self, entry_point_group: str = 'aiida.data', entry_point_name_filter=r'.*', **kwargs):
+    A command group using this class will automatically generate the sub commands from the entry points registered in
+    the given ``entry_point_group``. The entry points can be additionally filtered using a regex defined for the
+    ``entry_point_name_filter`` keyword. The actual command for each entry point is defined by ``command``, which should
+    take as a first argument the class that corresponds to the entry point. In addition, it should accept ``kwargs``
+    which will be the values for the options passed when the command is invoked. The help string of the command will be
+    provided by the docstring of the class registered at the respective entry point. Example usage:
+
+    .. code:: python
+
+        def create_instance(cls, **kwargs):
+            instance = cls(**kwargs)
+            instance.store()
+            echo.echo_success(f'Created {cls.__name__}<{instance.pk}>')
+
+        @click.group('create', cls=DynamicEntryPointCommandGroup, command=create_instance,)
+        def cmd_create():
+            pass
+
+    """
+
+    def __init__(self, command, entry_point_group: str = 'aiida.data', entry_point_name_filter=r'.*', **kwargs):
```

Comment:
```suggestion
    def __init__(self, command, entry_point_group: str, entry_point_name_filter=r'.*', **kwargs):
```
I think we don't need the default value set to `aiida.data` here?